Pinta.Core/Effects/HsvColor.cs

```diff
@@ -6,141 +6,66 @@
+			throw new ArgumentOutOfRangeException (nameof (sat), "must be in the range [0, 1]");
 
-		if (value < 0 || value > 100)
-			throw new ArgumentOutOfRangeException (nameof (value), "must be in the range [0, 100]");
+		if (val < 0 || val > 1)
+			throw new ArgumentOutOfRangeException (nameof (val), "must be in the range [0, 1]");
 
 		Hue = hue;
-		Saturation = saturation;
-		Value = value;
+		Sat = sat;
+		Val = val;
 	}
 
-	//        public static HsvColor FromColor(Color color)
-	//        {
-	//            RgbColor rgb = new RgbColor(color.R, color.G, color.B);
-	//            return rgb.ToHsv();
-	//        }
-	//
-	//        public Color ToColor()
-	//        {
-	//            RgbColor rgb = ToRgb();
-	//            return Color.FromArgb(rgb.Red, rgb.Green, rgb.Blue);
-	//        }
-
-	public readonly RgbColor ToRgb ()
-	{
-		// HsvColor contains values scaled as in the color wheel.
-		// Scale Hue to be between 0 and 360. Saturation
-		// and value scale to be between 0 and 1.
-		double h = (double) Hue % 360;
-		double s = (double) Saturation / 100;
-		double v = (double) Value / 100;
-
-		double r = 0;
-		double g = 0;
-		double b = 0;
-
-		if (s == 0) {
-			// If s is 0, all colors are the same.
-			// This is some flavor of gray.
-			r = v;
-			g = v;
-			b = v;
-		} else {
-			// The color wheel consists of 6 sectors.
-			// Figure out which sector you're in.
-			double sectorPos = h / 60;
-			int sectorNumber = (int) (Math.Floor (sectorPos));
+	public static HsvColor FromColor (Color c) => c.ToHsv ();
+	public Color ToColor (double alpha = 1) => Color.FromHsv (this);
 
-			// get the fractional part of the sector.
-			// That is, how many degrees into the sector
-			// are you?
-			double fractionalSector = sectorPos - sectorNumber;
+	public static HsvColor FromRgbD (double r, double g, double b) => new Color (r, g, b).ToHsv ();
 
-			// Calculate values for the three axes
-			// of the color. 
-			double p = v * (1 - s);
-			double q = v * (1 - (s * fractionalSector));
-			double t = v * (1 - (s * (1 - fractionalSector)));
+	public static HsvColor FromRgbI (int r, int g, int b) => new Color (r / 255.0, g / 255.0, b / 255.0).ToHsv ();
 
-			// Assign the fractional colors to r, g, and b
-			// based on the sector the angle is in.
-			switch (sectorNumber) {
-				case 0:
-					r = v;
-					g = t;
-					b = p;
-					break;
-
-				case 1:
-					r = q;
-					g = v;
-					b = p;
-					break;
-
-				case 2:
-					r = p;
-					g = v;
-					b = t;
-					break;
-
-				case 3:
-					r = p;
-					g = q;
-					b = v;
-					break;
-
-				case 4:
-					r = t;
-					g = p;
-					b = v;
-					break;
-
-				case 5:
-					r = v;
-					g = p;
-					b = q;
-					break;
-			}
-		}
-		// return an RgbColor structure, with values scaled
-		// to be between 0 and 255.
-		return new RgbColor ((int) (r * 255), (int) (g * 255), (int) (b * 255));
+	public static HsvColor FromBgra (ColorBgra c) => new Color (c.R / 255.0, c.G / 255.0, c.B / 255.0).ToHsv ();
```

Comment:
Minor suggestion: using `c.ToCairoColor()` would be good here just to avoid repeating the conversion logic. And similarly, `c.ToColorBgra()` when going in the other direction below